### PR TITLE
feat(tests): add implementation coverage tests for eip7883 modexp gas calculation

### PR DESCRIPTION
**Pull request by bshastry**
## 🗒️ Description

This PR adds a comprehensive set of implementation coverage (IC) test cases for the EIP-7883 ModExp precompile. These tests specifically target edge cases and implementation variations observed across different Ethereum clients.

### Key improvements:
- **Boundary Testing**: IC1, IC3, and IC4 test critical boundaries in the gas calculation formula where different implementations might diverge (32-byte boundary for complexity calculation, ceiling division edge cases)
- **Bit Counting Coverage**: IC5 and IC10 ensure all clients correctly count bits in sparse exponents, particularly when significant bits appear in unexpected positions
- **Native Library Edge Cases**: IC6 and IC7 test scenarios where clients might choose different optimization paths (native vs interpreted code), ensuring mathematical correctness is maintained
- **Error Handling**: IC9 validates consistent behavior for undefined operations (zero modulus)

### Implementation analysis:
The test cases were developed after analyzing the ModExp implementations in:
- **go-ethereum**: Uses native Go big.Int with special case optimizations
- **Nethermind**: Leverages GMP library with SIMD vector optimizations for input validation
- **Besu**: Dual implementation with native library selection based on input characteristics
- **Erigon**: Enhanced go-ethereum base with additional edge case handling
- **Reth/REVM**: Rust implementation with optional GMP backend

Each test case targets specific code paths that could lead to consensus failures if implemented differently across clients.

## 🔗 Related Issues or PRs

- Related to EIP-7883 implementation
- Complements existing ModExp test coverage in `test_modexp_thresholds.py`

## ✅ Checklist

- [x] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx --with=tox-uv tox -e lint,typecheck,spellcheck,markdownlint
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [x] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
- [ ] Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-spec-tests/blob/main/docs/writing_tests/post_mortems.md) to add an entry the list.
- [ ] Ported Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) or [tests/static](/ethereum/execution-spec-tests/blob/main/tests/static) have been assigned `@ported_from` marker.

### CODE REVIEW

File: tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py [hunk 1]
```diff
@@ -389,6 +406,14 @@ def create_modexp_variable_gas_test_cases():
...
     # │ E3  │  S   │  =  │  B   │ True  │   500   │ High bit in exponent last byte                │
     # │ E4  │  S   │  =  │  B   │ False │  4064   │ Maximum 32-byte exponent                      │
+    # │ IC1 │  L   │  =  │  B   │ True  │   500   │ Bit shift vs multiplication @ 33 bytes        │
+    # │ IC3 │  S   │  =  │  B   │ True  │   500   │ Ceiling division at 7 bytes                   │
+    # │ IC4 │  S   │  =  │  B   │ True  │   500   │ Ceiling division at 9 bytes                   │
+    # │ IC5 │  S   │  =  │  B   │ False │  2032   │ Bit counting in middle of exponent            │
+    # │ IC6 │  L   │  =  │  B   │ True  │   500   │ Native library even byte optimization         │
+    # │ IC7 │  L   │  =  │  B   │ True  │   500   │ Vector optimization 128-bit boundary          │
+    # │ IC9 │  S   │  =  │  B   │  N/A  │   N/A   │ Zero modulus handling                         │
+    # │ IC10│  S   │  =  │  B   │ False │  2032   │ Power-of-2 boundary with high bit             │
```
Comment (LouisTsai-Csie) [inline tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py:416]: I discover some mismatch here:

IC5: 2032 -> 2160
IC10: 2032 -> 4080

This is the result I get from logging `precompile_gas` fixture during test execution, is there anything wrong in our formula?

**Review by LouisTsai-Csie — CHANGES_REQUESTED** [review]

File: tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py (hunk 1 above)
Comment (LouisTsai-Csie) [reply tests/osaka/eip7883_modexp_gas_increase/test_modexp_thresholds.py]: I run the mismatch cases locally, and your revision is correct, please help me update the table please, thanks!

**Review by spencer-tb — APPROVED** [review]
Thanks again! 🙏
